### ISSUE TITLE
Building WDL-Forge sometimes produces a broken build

### ISSUE THREAD

**Issue by forkiesassds** [Pokechu22/WorldDownloader#175]
When I load into a singleplayer world the game crashes, and joining any multiplayer servers causes this error ```Internal Exception: org.spongeporwered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered```
Singleplayer crashlog: https://hasteb.in/ojiqujug.yaml

**Comment by Pokechu22** [comment]
This is SpongePowered/MixinGradle#9.  You should be able to build it properly if you use `clean build` instead of just `build`, in my experience.  I remember this also affecting the litemod in the past.  I should probably add a sanity check after building to make sure this doesn't happen.

**Comment by forkiesassds** [comment]
So I properly followed the building instructions, and now the game works now.

**Comment by Pokechu22** [comment]
Good to hear that the build process works fine (apart from this issue).  I'll keep this issue open until I fix the build problem.

**Comment by Tsoccerguy3** [comment]
![2020-06-02_20 41 04](https://user-images.githubusercontent.com/22108822/83583253-d84a5a80-a511-11ea-83b9-7f2b53ed67d6.png)

gradlew clean build compiles but the Forge builds dont load and produce the errors in picture

**Comment by Pokechu22** [comment]
Please read the message; there's a dependency you need to install (if you've got any suggestions on how to make it more obvious, other than including installation instructions in the forum post (which I plan to do), let me know).

**Comment by Tsoccerguy3** [comment]
![2020-06-03_07 27 32](https://user-images.githubusercontent.com/22108822/83631728-f93f9980-a56b-11ea-8905-dfa7e57c0538.png)

It's a good idea to post instructions including where to find the dependency. I have gradle extensions installed , but if there is something else , maybe add it to the compile instructions on the main page .
It complains about optifine but the same thing happens with or without optifine

**Comment by Pokechu22** [comment]
The dependency is [MixinBootstrap](https://www.curseforge.com/minecraft/mc-mods/mixinbootstrap), and it's linked in the error message on the last line.  It's not actually needed to build (or run it in the development environment), only to run it when you've installed it elsewhere.

**Comment by Tsoccerguy3** [comment]
[latest.log](https://github.com/Pokechu22/WorldDownloader/files/4728899/latest.log)

still not sure what this bootstrap is supposed to do but when its in the mod folder can not connect , see log

**Comment by forkiesassds** [comment]
> when its in the mod folder can not connect 

Try building it again using the same command line arguments.

**Comment by Tsoccerguy3** [comment]
should bootstrap jar be placed in the wdl folder some where for compile ?

**Comment by Pokechu22** [comment]
No; it's only needed at runtime (MixinBootstrap is what tells forge to load the mixins).  You have it in the right place based on the error you're getting (MixinBootstrap loaded, at least).

The issue you have in the log (`org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: @Shadow field world was not located in the target class net.minecraft.client.network.play.ClientPlayNetHandler. Using refmap mixins.ForgeWDL.refmap.json`) is the same one that this issue is about, and it's one I *thought* was resolved by using `clean build`.  If `clean build` doesn't work, try first using `clean`, and then using `build`.

**Comment by Tsoccerguy3** [comment]
Hi Poke I tried multiple times to generate a Forge build , with no luck . I've followed your advise but no luck
[debug.log](https://github.com/Pokechu22/WorldDownloader/files/4755173/debug.log)

**Comment by Tsoccerguy3** [comment]
To be clear it builds with no errors , but can not log on to a server

**Comment by Pokechu22** [comment]
Try changing this line:

https://github.com/Pokechu22/WorldDownloader/blob/5b713b6ef3992eb1ce176c9a56c6c1fa4c4a7d35/1.15.2-forge/src/main/java/wdl/forge/mixin/MixinNetHandlerPlayClient.java#L36

to `@Shadow(aliases={"f", "field_147300_g"})` &mdash; that may help (but it really should not be needed)

**Comment by Tsoccerguy3** [comment]
That worked

**Comment by Pokechu22** [comment]
Well, I don't really have any idea why it's not remapping, so I'll just apply that workaround to all versions (this same thing has happened to me with the litemod in the past).
